lib/matplotlib/image.py

```diff
@@ -985,6 +985,15 @@ def set_data(self, x, y, A):
                     self.is_grayscale = True
             else:
                 raise ValueError("3D arrays must have RGB or RGBA as last dim")
+
+        # For efficient cursor readout, ensure x and y are increasing.
+        if x[-1] < x[0]:
```

Comment:
No, we have never enforced monotonic x or y for any of the pcolor-like functions; that has implicitly been left to the user.  If the user supplies non-monotonic x and/or y the plotted results will be odd, and with this PR as-is the cursor readout will sometimes be hard to predict--but the plot won't make sense anyway, so I don't see that it matters.  So it seems to me that you are raising a question that is independent of this PR--should we add more input validation for this category of plot?